clients/src/test/java/org/apache/kafka/clients/admin/internals/AdminFetchMetricsManagerTest.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin.internals;
+
+import org.apache.kafka.common.MetricName;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.Metrics;
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.common.metrics.stats.Avg;
+import org.apache.kafka.common.metrics.stats.Max;
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class AdminFetchMetricsManagerTest {
+    private static final double EPSILON = 0.0001;
+
+    private final Time time = new MockTime(1, 0, 0);
+    private Metrics metrics;
+    private AdminFetchMetricsManager adminFetchMetricsManager;
+
+
+    @BeforeEach
+    public void setup() {
+        metrics = new Metrics(time);
+        adminFetchMetricsManager = new AdminFetchMetricsManager(metrics);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        if (metrics != null) {
+            metrics.close();
+            metrics = null;
+        }
+
+        adminFetchMetricsManager = null;
+    }
+
+    @Test
+    public void testSingleNodeLatency() {
+        String connectionId = "0";
+        MetricName nodeLatencyAvg = metrics.metricName("request-latency-avg", "group");
+        MetricName nodeLatencyMax = metrics.metricName("request-latency-max", "group");
```

Comment:
nit: Maybe we should consider extracting the string literal "group" as a variable since it is used everywhere.